skxray/img_proc/threshops.py

```diff
@@ -0,0 +1,324 @@
+# Module for the BNL image processing project
+# Developed at the NSLS-II, Brookhaven National Laboratory
+# Developed by Gabriel Iltis, Oct. 2013
+"""
+This module contains tools for thresholding data sets.
+"""
+
+"""
+REVISION LOG: (FORMAT: "PROGRAMMER INITIALS: DATE -- RECORD")
+GCI: 2/11/2014 -- Modifying documentation of the package functions for 
+    inclusion in the bulk module pull to GITHUB
+GCI: 2/19/2014 -- Renaming module from C4_threshops to threshops.py
+    Updating documentation to docstring convention.
+    Changed module back from class structure to a simple module containing
+    thresholding function definitions.
+GCI: 9/23/14 -- Modified all functions to auto-wrap into the image
+    processing library in vistrails.
+"""
+
+import numpy as np
+import skimage.filter as sk
+
+
+def thresh_globalGT(src_data, thresh_value, md_dict=None):
+    """
+    This function is a basic thresholding operation allowing for isolation, or 
+    selection, of all voxels having an intensity value greater than the 
+    specificed threshold value. The result is a binary array of equal size 
+    containing voxel values of either 0 or 1.
+
+    Parameters
+    ----------
+    src_data : ndarray
+        Specify the volume to be thresholded
+    
+    thresh_value : float
+        Specify the threshold value, above which contains the desired voxel 
+        values to be isolated. Input value is of type float, however,
+        the dtype of this parameter will be automatically converted to the
+        dtype of the source data if the source data is not of type float.
+
+    md_dict : dict, optional
+        Metadata dictionary for the data set(s) being segmented or analyzed,
+        or for the analysis being conducted
+
+    Returns
+    -------
+    output : ndarray
+        The function returns a binary array where all voxels with values equal 
+        to 1 correspond to voxels within the identified threshold range.
+    """
+
+    if thresh_value > np.amax(src_data):
+        raise ValueError("Selected threshold value is greater than the "

```

Comment:
The rough usage is:

    images = thresh_globalGT(images, number)

A second function could do:

   list_of_nonblack_counts = check_saturation(images)

If everything is blacked out in some frame, one of the elements of the returned list will be 0. E.g.,
    >>> check_saturation(thresholded_images)
    [100, 50, 0, 25, 1]

The thresholding functions like `thresh_globalGT` should issue a warning stating that at least one element of the Z stack is fully black (or, hey, white), and it can direct the user to `check_saturation` to explore further.